the-blue-alliance-ios/AppDelegate.swift

```diff
@@ -13,105 +13,58 @@ let kNoSelectionNavigationController = "NoSelectionNavigationController"
+            fatalError("window should be window")
         }
+        window.rootViewController = launchViewController()
+        window.makeKeyAndVisible()
 
-        setupAppearance()
-
+        // Setup our Firebase app - make sure this is called before other Firebase setup
         FirebaseApp.configure()
-        // Setup Remote Config
-        RemoteConfig.setupRemoteConfig()
-
-        // Assign our Push Notification delegates
-        pushService = PushService(userDefaults: UserDefaults.standard,
-                                  myTBA: MyTBA.shared,
-                                  retryService: RetryService())
-        Messaging.messaging().delegate = pushService
-        UNUserNotificationCenter.current().delegate = pushService
 
-        // Setup our React Native service
-        let reactNativeService = ReactNativeService(userDefaults: UserDefaults.standard,
-                                                    fileManager: FileManager.default,
-                                                    firebaseStorage: Storage.storage(),
-                                                    firebaseOptions: FirebaseOptions.defaultOptions()!,
-                                                    retryService: RetryService())
-        reactNativeService.registerRetryable(initiallyRetry: true)
+        // TODO: Load this from some secrets file
+        TBAKit.sharedKit.apiKey = "OHBBu0QbDiIJYKhAedTfkTxdrkXde1C21Sr90L1f1Pac4ahl4FJbNptNiXbCSCfH"
 
-        // myTBA/Google Sign In
-        GIDSignIn.sharedInstance().clientID = FirebaseApp.app()?.options.clientID
-        GIDSignIn.sharedInstance().delegate = self
+        // Assign our Push Service as a delegate to all push-related classes
+        AppDelegate.setupPushServiceDelegates(with: pushService)
 
-        // If we're authenticated with Google but don't have a Firebase user, get a Firebase user
-        if GIDSignIn.sharedInstance().hasAuthInKeychain() && Auth.auth().currentUser == nil {
-            GIDSignIn.sharedInstance().signInSilently()
+        // Setup our React Native service
+        AppDelegate.setupReactNativeService()
+
+        // Kickoff background myTBA/Google sign in, along with setting up delegates
+        setupGoogleAuthentication()
+
+        // Setup our remote config - to be used by our app setup operation
+        let remoteConfigService = RemoteConfigService(remoteConfig: RemoteConfig.remoteConfig(),
+                                                      retryService: RetryService())
+        remoteConfigService.registerRetryable()
+
+        // Our app setup operation will load our persistent stores, fetch our remote config, propogate persistance container
+        let appSetupOperation = AppSetupOperation(window: window,
+                                                  rootSplitViewController: rootSplitViewController(),
+                                                  persistentContainer: NSPersistentContainer(name: "TBA"),
```

Comment:
We should think about this one... should we keep a root strong reference to this in the app delegate? It probably doesn't matter since all of the root view controllers have a strong reference but